### PR TITLE
Feature/zaak archiving

### PR DESCRIPTION
**Pull request by stevenbal**
fixes https://github.com/open-zaak/open-zaak/issues/257

I will also update the newman tests for zaak archiving

### CODE REVIEW

File: src/openzaak/components/zaken/brondatum.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from datetime import date, datetime
 from typing import Union
 
-from django.db.models import Max
+from django.db.models import Max, Min
```
Comment (sergei-maertens) [inline src/openzaak/components/zaken/brondatum.py]: unused `Min` import?

File: src/openzaak/components/zaken/brondatum.py [hunk 2]
```diff
@@ -187,7 +196,16 @@ def get_brondatum(
...
-        # TODO: Relation from Zaak to Besluit is not implemented yet...
-        raise NotImplementedError
+        zaakbesluiten = zaak.besluit_set.all()
+        if not zaakbesluiten.exists():
+            # Cannot use vervaldatum_besluit if Zaak has no Besluiten
+            raise DetermineProcessEndDateException(
+                _("Geen besluiten aan zaak gekoppeld om brondatum uit af te leiden.")
+            )
+
+        max_vervaldatum = zaakbesluiten.aggregate(Max("vervaldatum"))[
```
Comment (sergei-maertens) [inline src/openzaak/components/zaken/brondatum.py]: Can you add a test where there _does_ exist a `ZaakBesluit`, BUT the `vervaldatum` is `None`? That should return a validation error saying the vervaldatum must be set first or something like that?

@joeribekker sanity check that I'm not talking nonsense?

Comment (joeribekker) [reply src/openzaak/components/zaken/brondatum.py]: It can be `None` so thats a valid case.
